unit-tests/func/hw-errors/l500/l500-error-common.h

```diff
@@ -55,45 +55,41 @@ void trigger_error_or_exit( const rs2::device & dev, uint8_t opcode, uint8_t num
     }
 }
 
+// This function loop over all posiable hw errors use hw-monitor command to trigger them
+// and check that we call to user callback with the error
 void validate_errors_handling( const rs2::device & dev,
-                               const std::map< uint8_t, std::string > & error_report)
+                               const std::map< uint8_t, std::string > & error_report )
 {
-    std::vector< sensor > list;
-    REQUIRE_NOTHROW( list = dev.query_sensors() );
-    REQUIRE( list.size() > 0 );
+    auto depth_sens = dev.first< rs2::depth_sensor >();
 
-    for( auto && subdevice : list )
-    {
-        std::string notification_description;
-        rs2_log_severity severity;
-        std::condition_variable cv;
-        std::mutex m;
+    REQUIRE( depth_sens.supports( RS2_OPTION_ERROR_POLLING_ENABLED ) );
 
-        if( subdevice.supports( RS2_OPTION_ERROR_POLLING_ENABLED ) )
-        {
-            subdevice.set_notifications_callback( [&]( rs2::notification n ) {
-                std::unique_lock< std::mutex > lock( m );
+    std::string notification_description;
+    rs2_log_severity severity;
+    std::condition_variable cv;
+    std::mutex m;
 
-                notification_description = n.get_description();
-                severity = n.get_severity();
-                cv.notify_one();
-            } );
 
-            REQUIRE_NOTHROW( subdevice.set_option( RS2_OPTION_ERROR_POLLING_ENABLED, 1 ) );
+    depth_sens.set_notifications_callback( [&]( rs2::notification n ) {
```

Comment:
What happens after we exit this function? Does this notification still get called? If so, isn't it dangerous?

Reply:
fixed added depth_sens.set_option(RS2_OPTION_ERROR_POLLING_ENABLED, 0); at the end